src/project-root.ts

```diff
@@ -21,20 +21,17 @@ export function getProjectWsFolder(): vscode.WorkspaceFolder {
 }
 
 export async function findProjectRootUri(
-    projectFileNames,
-    doc,
-    workspaceFolder
-): Promise<Object> {
-    let searchUri = doc?.uri || workspaceFolder?.uri;
+    projectFileNames: string[],
+    doc: vscode.TextDocument,
+    workspaceFolder: vscode.WorkspaceFolder
+): Promise<vscode.Uri> {
+    let searchUri = doc.uri || workspaceFolder.uri;
```

Comment:
won't this throw if doc is null?

Reply:
You're right! It can! I was trusting the types but it turns on we've disabled strictNullChecks which is honestly kind of horrifying. I turned them it on and got 735 errors 🙀. Not knowing where anything can be null/undefined has to lead to a ton of issues. oof

Reply:
@PEZ we might want to turn on strictNullChecks https://www.typescriptlang.org/tsconfig#strictNullChecks -- it'll keep us safe from nulls and undefined, which is a huge pain in js otherwise.

Reply:
It is causing all sorts of problems in Calva. And sounds like there are 735 potential problems there...

Reply:
yay, another PR!! or adding it to this one because why not?

Reply:
I upgraded the vscode types to try to fix null issues before noticing strict null checks was off and I got this error: 

https://app.circleci.com/pipelines/github/BetterThanTomorrow/calva/4427/workflows/430f6aad-a2ea-4592-b720-634506ce0fa4/jobs/18002/parallel-runs/0/steps/0-109

But it seems like the newest version is 1.64.0?

https://code.visualstudio.com/updates/v1_64

Reply:
I think both of those things will help, but it might make sense to do it in a separate PR. Fixing 733 errors in here might be a bit much 😄

Reply:
I second a separate PR, but that will be awesome.